### ISSUE TITLE
Fix  tuple concurrently updated error when concurrently calling alter_job_schedule

### ISSUE THREAD

**Issue by cevian** [timescale/timescaledb#2165]
**Relevant system information:**
 - OS: [e.g. Ubuntu 16.04, Windows 10 x64, etc] MacOS
 - PostgreSQL version (output of `postgres --version`): 12
 - TimescaleDB version (output of `\dx` in `psql`): [e.g. 1.0.0]. 1.7.x
 - Installation method: [e.g., "using Docker", "apt install", "source"]: source -- shouldn't matter

Calling alter_job_schedule for the same job can lead to `tuple concurrently updated`. Since those
errors should never be user-visible, we need to fix that by adding a stronger lock. Sample log output below:

```
2020-07-30 11:44:21.595 EDT [9714] ERROR:  tuple concurrently updated
2020-07-30 11:44:21.595 EDT [9714] STATEMENT:  SELECT alter_job_schedule(
								(SELECT job_id
								FROM _timescaledb_config.bgw_policy_compress_chunks p
								INNER JOIN _timescaledb_catalog.hypertable h ON (h.id = p.hypertable_id)
								WHERE h.schema_name = $1 and h.table_name = $2),
								next_start=>$3)```

**Comment by k-rus** [comment]
@cevian Do you have a way to reproduce it?

**Comment by k-rus** [comment]
This issue was fixed in 2.0 by https://github.com/timescale/timescaledb/pull/2689

It is not critical to fix in 1.7, since it can be re-run in the case of failure.
